www/docs/server/infer-types.md

```diff
@@ -133,12 +143,21 @@ export type RouterInputs = inferRouterInputs<AppRouter>;
 export const trpc = createTRPCReact<AppRouter>();
 
 // @filename: usePostCreate.ts
-import { ReactQueryOptions, trpc } from './trpc';
+// @noErrors
+import { type ReactQueryOptions, trpc } from './trpc';
 
 type PostCreateOptions = ReactQueryOptions['post']['create'];
 
 function usePostCreate(options?: PostCreateOptions) {
-  return trpc.post.create.useMutation(options);
+  const utils = trpc.useContext();
+  return trpc.post.create.useMutation({
+    onSuccess(post) {         
```

Comment:
not sure why but couldn't get shake working here, got implicit any so had to smash a `@noErrors` on it too :/